### PR TITLE
Refactor of media player unit test

### PR DESCRIPTION
**Pull request by HarveyPeachey**
Related to https://github.com/bbc/simorgh/pull/4256

**Overall change:** Refactor of Media player unit tests to keep consistency with existing [`ArticleFigure`](https://github.com/bbc/simorgh/tree/latest/src/app/containers/ArticleFigure) unit tests for clarity and ease of use.

**Code changes:**

- Moved context provider to `fixtureData.jsx`.
- Exported relevant blocks for metadata tests.
- Created a `GenerateFixtureData` wrapper to help generate fixture data for tests with propType checking.
- Moved all test logic to `fixtureData.jsx`.
---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/MediaPlayer/index.test.jsx [hunk 1]
```diff
@@ -1,106 +1,32 @@
...
+import {
+  VideoFixture,
+  VideoAmpFixture,
+  VideoFixtureNoPlaceHolder,
+  VideoFixtureNoVersionId,
+  VideoFixtureToggledOff,
+} from './fixtureData';
 
 describe('MediaPlayer', () => {
   describe('is called correctly', () => {
```
Comment (jamesdonoh) [inline src/app/containers/MediaPlayer/index.test.jsx]: I think this describe block is a bit redundant, it doesn't add much descriptive value.

File: src/app/containers/MediaPlayer/index.test.jsx [hunk 2]
```diff
@@ -1,106 +1,32 @@
...
-
-ContextWrapper.defaultProps = {
-  toggleState: defaultToggles,
-};
+import {
+  VideoFixture,
+  VideoAmpFixture,
+  VideoFixtureNoPlaceHolder,
+  VideoFixtureNoVersionId,
+  VideoFixtureToggledOff,
```
Comment (jamesdonoh) [inline src/app/containers/MediaPlayer/index.test.jsx]: 'Fixture' seems redundant info. Also it's not just Video is it? And I think we could be explicit about the canonical ones. What about naming like:
```
MediaPlayerCanonical
MediaPlayerCanonicalNoPlaceholder
MediaPlayerAmp
```
etc.? Open to other suggestions.

File: src/app/containers/MediaPlayer/fixtureData.jsx [hunk 3]
```diff
@@ -279,3 +288,97 @@ export const missingVpidFixture = [
...
+    id="foo"
+    pageType="article"
+    pathname="/pathname"
+  >
+    <ServiceContextProvider service="news">
+      <ToggleContext.Provider
+        value={{ toggleState, toggleDispatch: jest.fn() }}
+      >
+        <MediaPlayerContainer
+          blocks={Array.isArray(blocks) ? blocks : [blocks]}
```
Comment (jamesdonoh) [inline src/app/containers/MediaPlayer/fixtureData.jsx]: When might `blocks` be a single block?

Comment (HarveyPeachey) [reply src/app/containers/MediaPlayer/fixtureData.jsx]: Single block: https://github.com/bbc/simorgh/blob/ce04f0f1262c3080029a033832ef2b028c70f7e0/src/app/containers/MediaPlayer/fixtureData.jsx#L50
Multiple blocks: https://github.com/bbc/simorgh/blob/ce04f0f1262c3080029a033832ef2b028c70f7e0/src/app/containers/MediaPlayer/fixtureData.jsx#L244

File: src/app/containers/MediaPlayer/index.test.jsx (hunk 2 above)
Comment (HarveyPeachey) [reply src/app/containers/MediaPlayer/index.test.jsx]: Before the refactor there were only tests for the video player and not the audio player so I kept it consistent. Was thinking of creating a new issue to add them in.

File: src/app/containers/MediaPlayer/index.test.jsx (hunk 1 above)
Comment (HarveyPeachey) [reply src/app/containers/MediaPlayer/index.test.jsx]: Yep I agree, this was just the naming of it before the refactor. How about changing it to `is rendered with` and then each snapshot test will mention what the media player is rendered with or do you suggest completely removing this describe block?

File: src/app/containers/MediaPlayer/fixtureData.jsx (hunk 3 above)
Comment (jamesdonoh) [reply src/app/containers/MediaPlayer/fixtureData.jsx]: I'd leave this to the caller to minimise confusion over types.

**Review by jamesdonoh — APPROVED** [review]
Nice work 👍

**Review by 12 — APPROVED** [review]
LGTM 👍

**Pull request merged**